semgrep/semgrep/semgrep_main.py

```diff
@@ -385,14 +385,31 @@ def main(
         logger.info(f"Running baseline scan with base set to: {baseline_commit}")
         try:
             with baseline_handler.baseline_context():
+                # Need to reinstantiate target_manager since
+                # filesystem has changed
+                try:
+                    baseline_target_manager = TargetManager(
```

Comment:
Maybe this should be a method on the TargetManager? Something like `target_manager.reset()`?

Reply:
although reset() implies that it's an in-place operation so that's not ideal